aws_generated_data/commands/rds_eol.py

```diff
@@ -0,0 +1,150 @@
+        return str(self)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Engine):
+            return False
+        return (self.name, self.url) == (other.name, other.url)
+
+
+def engine_with_url(value: str) -> Engine:
+    return Engine(value)
+
+
+def parse_date(date_str: str) -> datetime:
+    try:
+        return datetime.strptime(date_str, "%B %Y")
+    except ValueError:
+        return datetime.strptime(date_str, "%d %B %Y")
+
+
+def parse_aws_release_calendar(page: str) -> list[CalItem]:
+    items: list[CalItem] = []
+    soup = BeautifulSoup(page, "html.parser")
+    version_table = soup.find_all("table")[0]
+    for row in version_table.find_all("tr"):
+        cols = row.find_all("td")
+        if len(cols) == 4:
+            items.append((cols[0].text.strip(), parse_date(cols[3].text.strip())))
+
+    return items
+
+
+def get_rds_eol_data(engine: Engine) -> list[RdsItem]:
+    version_page = requests.get(engine.url)
+    return [
+        RdsItem(engine=engine.name, version=version, eol=d.date())
+        for version, d in parse_aws_release_calendar(version_page.text)
+    ]
+
+
+def read_output_file(output: Path) -> list[RdsItem]:
+    try:
+        return [RdsItem(**item) for item in yaml.safe_load(output.read_text())]
+    except (TypeError, FileNotFoundError, ValidationError):
+        log.warning(f"Failed to load {output}")
+        return []
+
+
+def write_output_file(output: Path, rds_items: list[RdsItem]) -> None:
+    log.info(f"Saving to {output} ...")
+    output.write_text(
+        yaml.dump(
+            RdsEol(rds_items).model_dump(),
+            explicit_start=True,
+            indent=2,
+            default_flow_style=False,
+        )
+    )
+
+
+def filter_rds_items(rds_items: Iterable[RdsItem], expired_date: date) -> list[RdsItem]:
+    return [item for item in rds_items if item.eol > expired_date]
+
+
+@app.command()
+def fetch(
+    engines: Annotated[
+        list[Engine],
+        typer.Option(
+            parser=engine_with_url,
+            envvar="AGD_RDS_EOL_ENGINES",
+            help="Engines to sync; format: engine_name:release_calendar_url",
+        ),
+    ],
+    output: Annotated[
+        Path,
+        typer.Option(
+            help="Output file",
+            envvar="AGD_RDS_EOL_OUTPUT",
+        ),
+    ],
+    clean_up_days: Annotated[
+        int,
+        typer.Option(
+            help="Remove items older than this number of days",
+            envvar="AGD_RDS_CLEAN_UP_DAYS",
+        ),
+    ] = 365,
+) -> None:
+    """Fetch RDS EOL data from AWS and saves it to a file."""
+    rds_items = read_output_file(output)
+    for engine in engines:
+        log.info(f"Processing {engine} ...")
+        rds_items += get_rds_eol_data(engine)
+
+    rds_items = filter_rds_items(
+        rds_items,
+        expired_date=date.today() - timedelta(days=clean_up_days),
+    )
+    write_output_file(output, sorted(set(rds_items), reverse=True))
```

Comment:
What if some `eol` data changed from last time? Will both old eol and new eol saved to output file?